src/hotspot/share/gc/g1/g1ConcurrentRefineStats.hpp

```diff
@@ -33,34 +33,56 @@
 // Used for collecting per-thread statistics and for summaries over a
 // collection of threads.
 class G1ConcurrentRefineStats : public CHeapObj<mtGC> {
-  Tickspan _refinement_time;
-  size_t _refined_cards;
-  size_t _precleaned_cards;
-  size_t _dirtied_cards;
+  jlong _sweep_duration;              // Time spent sweeping the table finding non-clean cards
+                                      // and refining them.
+  jlong _yield_duration;              // Time spent yielding during the sweep (not doing the sweep).
+
+  size_t _cards_scanned;              // Total number of cards scanned.
+  size_t _cards_clean;                // Number of cards found clean.
+  size_t _cards_not_parsable;         // Number of cards we could not parse and left unrefined.
+  size_t _cards_still_refer_to_cset;  // Number of cards marked still young.
```

Comment:
`_cards_still_refer_to_cset` from the naming it is not clear what the difference is with `_cards_refer_to_cset`, the comment is not helping with that

Reply:
`cards_still_refer_to_cset` refers to cards that were found to have already been marked as `to-collection-set`. Renamed to `_cards_already_refer_to_cset`, would that be okay?

Reply:
Fwiw, this is just for statistics, so if you want I can remove these. I did some experiments with re-examining these cards too to see whether we could clear them later. For determining if/when to do that a rate of increase for the young cards has been interesting.

As mentioned, if you want I can remove them.

Reply:
`_cards_already_refer_to_cset` is fine by me, i don't like the option of removing them